app/selectors/autocomplete.js

```diff
@@ -242,3 +242,22 @@ export const makeGetMatchTermForDateMention = () => {
         return lastMatchTerm;
     };
 };
+
+export const getDeletedPublicChannelsIds = createSelector(
+    getMyChannels,
+    getOtherChannels,
+    (myChannels, otherChannels) => {
+        const channels = myChannels.filter((c) => {
+            return (c.type === General.OPEN_CHANNEL);
+        }).concat(otherChannels);
+
+        return new Set(channels.filter((c) => c.delete_at !== 0).map((c) => c.id));
```

Comment:
I think is best to use a `channels.reduce` instead of filter and then map, you can accomplish the same thing by iterating once instead of twice the same array.

An you could remove the `myChannels.filter` to and do the same thing on the reduce callback, so in the end you iterate over the whole thing once instead of a total of three times